Makefile

```diff
@@ -171,7 +177,22 @@ build-tf24-gpu: build-gpu-cuda-111-base
 
 .PHONY: build-tf2-cpu
 build-tf2-cpu: build-cpu-py-38-base
-	docker build -f Dockerfile-default-cpu \
+ifeq ($(NGC_PUBLISH),)
```

Comment:
could we have `-t` flags to be conditional, instead of the whole command-line, to minimize a chance someone forgets to update a value? i.e. in pseudocode

ifeq ($(NGC_PUBLISH),)
  TFLAGS := """		-t $(DOCKERHUB_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH) \
		-t $(DOCKERHUB_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION) \
		-t $(DOCKERHUB_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION) \
		-t $(NGC_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH) \
		-t $(NGC_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH) \
		-t $(NGC_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION) \
		-t $(NGC_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION) \"""
else
  TFLAGS := """
	-t $(DOCKERHUB_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH) \
		-t $(DOCKERHUB_REGISTRY)/$(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION) \
  """
endif 
docker buildx build -f Dockerfile-default-cpu \
	    --platform linux/arm64,linux/amd64 \
		--build-arg BASE_IMAGE="$(DOCKERHUB_REGISTRY)/$(CPU_PY_38_BASE_NAME)-$(SHORT_GIT_HASH)" \
		--build-arg TENSORFLOW_PIP="tensorflow-cpu==2.8.2" \
		--build-arg TORCH_PIP="torch==1.10.2+cpu torchvision==0.11.3+cpu torchaudio==0.10.2+cpu -f https://download.pytorch.org/whl/cpu/torch_stable.html" \
		--build-arg LIGHTNING_PIP="pytorch_lightning==1.5.10 torchmetrics==0.5.1" \
		--build-arg TORCH_PROFILER_GIT="https://github.com/pytorch/kineto.git@7455c31a01dd98bd0a863feacac4d46c7a44ea40" \
		--build-arg HOROVOD_PIP="horovod==0.24.2" \
        $(TFLAGS) \
		--push \
		.

Reply:
I managed to create conditionals for image tags along the lines you suggested, and it works in dev mode. It should also work in production mode unless there is some problem with the line length.

In dev mode, `CPU_TF27_TAGS` resolves to
`-t determinedai/environments-dev:py-3.8-tf-2.7-cpu-9d4ef7c -t determinedai/environments-dev:py-3.8-tf-2.7-cpu-0.18.4 \`, and it works.

In prod mode, it will resolve to something like
`-t determinedai/environments-dev:py-3.8-tf-2.7-cpu-9d4ef7c -t determinedai/environments-dev:py-3.8-tf-2.7-cpu-0.18.4 -t nvcr.io/isv-ngc-partner/determined/environments-dev:py-3.8-tf-2.7-cpu-9d4ef7c -t nvcr.io/isv-ngc-partner/determined/environments-dev:py-3.8-tf-2.7-cpu-0.18.4`
which is a bit long. I think the limit for `dash` is something like 8000 bytes so we should be fine.